src/code_28910_len_5090.c

```diff
@@ -1,16 +1,46 @@
-#include "common.h"
+#include "code_28910_len_5090.h"
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004D510);
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004D794);
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004D7E0);
 
-INCLUDE_ASM(s32, "code_28910_len_5090", func_8004DA0C);
+s32 func_8004DA0C(UNK_TYPE arg0) {
```

Comment:
this function may be able to be rewritten (potentially early return on the initial check of arg0)

Reply:
I wasn't able to get this to match without a `ret` var- the asm stores in s0 then jumps to `move v0, s0; jr ra` which is weird